tests/test_io.py

```diff
@@ -240,6 +240,21 @@ def test_sdf_props_and_conformer_preserved(tmp_path):
     assert conf.Is3D()
     np.testing.assert_almost_equal(conf.GetPositions(), pos, decimal=4)
 
+def test_read_mol2(datadir):
```

Comment:
Can you add a case where the reader is supposed to fail. The mol2 file definition is very ambiguous, so this might be hard but I think a negative test case would be nice.

Reply:
ok! Let me work on that.

Reply:
So I added cases where some mol2s will return `None` due to manual mutations to the mol2 file. I added these manually mutated mol2s into `temp.mol2`. Further, I added a test case where it will return an `ValueError` exception when  `fail_if_invalid=True` is an input parameter.